Float.Core/UX/NavigationPageNavigationContext.cs

```diff
@@ -104,6 +105,45 @@ public void DismissPage(bool animated = true)
             });
         }
 
+        /// <inheritdoc />
+        public async Task DismissPageAsync(bool animated = true)
+        {
+            if (HasModal)
```

Comment:
@EBusch I noticed this was inside `DeviceProxy.BeginInvokeOnMainThread` for the original method. Is this thread safe? (or does it not matter)?